standard/expressions.md

```diff
@@ -586,8 +588,18 @@ The corresponding parameters for function member arguments are established as fo
 
--   For a value parameter, the argument expression is evaluated and an implicit conversion ([§11.2](conversions.md#112-implicit-conversions)) to the corresponding parameter type is performed. The resulting value becomes the initial value of the value parameter in the function member invocation.
--   For a reference or output parameter, the variable reference is evaluated and the resulting storage location becomes the storage location represented by the parameter in the function member invocation. If the variable reference given as a reference or output parameter is an array element of a *reference_type*, a run-time check is performed to ensure that the element type of the array is identical to the type of the parameter. If this check fails, a `System.ArrayTypeMismatchException` is thrown.
+-   For a value parameter, if the parameter’s passing mode is value
+  -  the argument expression is evaluated and an implicit conversion ([§11.2](conversions.md#112-implicit-conversions)) to the corresponding parameter type is performed. The resulting value becomes the initial value of the value parameter in the function member invocation.
+  -  otherwise, the parameter’s passing mode is input, in which case, an unnamed *variable_reference* is created with the same type as that of the corresponding parameter. The argument expression is evaluated and an implicit conversion ([§11.2](conversions.md#112-implicit-conversions)) to the corresponding parameter type is performed. The resulting value becomes the initial value of the unnamed *variable_reference* whose storage location becomes the storage location represented by the input parameter in the function member invocation.  
+     > *Example*: Given the following declarations and method calls:
+     > ```csharp
+     > public static void M1(in int p1) { … }
+     > int i = 10;
+     > M1(i);         // transformed to a temporary input argument
+     > M1(i + 5);     // transformed to a temporary input argument
+     > ```
+     > In each method call, an unnamed `int` is created, initialized with the argument’s value, and then passed as an input argument. See [§12.6.4.2](expressions.md#12642-applicable-function-member) and §better-argument-passing-mode-new-clause. *end example*  
+-   For an input, output, or reference parameter, the variable reference is evaluated and the resulting storage location becomes the storage location represented by the parameter in the function member invocation. If the variable reference given as an input, output, or reference is an array element of a *reference_type*, a run-time check is performed to ensure that the element type of the array is identical to the type of the parameter. If this check fails, a `System.ArrayTypeMismatchException` is thrown.
```

Comment:
The run-time check should not be performed for an input parameter. See <https://github.com/dotnet/csharpstandard/issues/264#issuecomment-874738023>.

Reply:
Updated